src/app/page.tsx

```diff
@@ -99,16 +87,15 @@ export default function Home() {
         result += text;
 
         setMessages((prevMessages) => {
-          const lastMessage = prevMessages[prevMessages.length - 1]; // Get the last message (assistant's placeholder)
-          const otherMessages = prevMessages.slice(0, -1); // Get all other messages
+          const otherMessages = prevMessages.slice(0);
```

Comment:
add back the 
```typescript
         const lastMessage = prevMessages[prevMessages.length - 1]; // Get the last message (assistant's placeholder)
         const otherMessages = prevMessages.slice(0, -1); // Get all other messages
```
instead of only the
```typescript
 const otherMessages = prevMessages.slice(0);
 ```